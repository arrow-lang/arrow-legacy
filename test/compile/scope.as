def main() {
  if false {
    let mutable index = 10;
    index += 10;
  }

  while true {
    let mutable counter = 20;
    counter -= 1;
  }
}
