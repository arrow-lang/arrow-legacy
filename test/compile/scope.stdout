define internal void @scope..init() {
  ret void
}

define internal void @scope.main() {
  br i1 false, label %select-then, label %select-next

select-then:                                      ; preds = %0
  %index = alloca i128
  store i128 10, i128* %index
  %1 = load i128* %index
  %2 = add i128 %1, 10
  store i128 %2, i128* %index
  br label %select-next

select-next:                                      ; preds = %select-then, %0
  br label %loop-cond

loop-cond:                                        ; preds = %loop-body, %select-next
  br i1 true, label %loop-body, label %loop-merge

loop-body:                                        ; preds = %loop-cond
  %counter = alloca i128
  store i128 20, i128* %counter
  %3 = load i128* %counter
  %4 = sub i128 %3, 1
  store i128 %4, i128* %counter
  br label %loop-cond

loop-merge:                                       ; preds = %loop-cond
  ret void
}

define i32 @main(i32, i8**, i8**) {
  call void @scope..init()
  call void @scope.main()
  ret i32 0
}
